mm2src/rpc/lp_commands.rs

```diff
@@ -190,7 +191,15 @@ pub async fn my_balance(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Stri
     Ok(try_s!(Response::builder().body(res)))
 }
 
-pub fn stop(ctx: MmArc) -> HyRes {
+pub async fn stop(ctx: MmArc) -> Result<Response<Vec<u8>>, String> {
+    let dispatcher_ctx = DispatcherContext::from_ctx(&ctx).unwrap();
+    let event: StopCtxEvent = StopCtxEvent {};
+    dispatcher_ctx
+        .dispatcher
+        .read()
+        .await
+        .dispatch_async(ctx.clone(), event.into())
+        .await;
```

Comment:
We call the `MmCtx::stop` method from another one place: on the extern function [mm2_stop](https://github.com/KomodoPlatform/atomicDEX-API/blob/v1.1.0/mm2src/mm2_lib/mm2_native_lib.rs#L236).

I suggest using `MmCtx::on_stop` to dispatch `StopCtxEvent` like it does [spawn_rpc](https://github.com/KomodoPlatform/atomicDEX-API/blob/v1.1.0/mm2src/rpc.rs#L276).

Reply:
Is the function `MmCtx::on_stop` is called before the member of the structure are destroyed? In this case that's probably a good idea!